core/src/util/usProperties.cpp

```diff
@@ -0,0 +1,130 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/saschazelzer/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "usProperties_p.h"
+
+#include <limits>
+#include <stdexcept>
+#ifdef US_PLATFORM_WINDOWS
+#include <string.h>
+#define ci_compare strnicmp
+#else
+#include <strings.h>
+#define ci_compare strncasecmp
+#endif
+
+namespace us {
+
+Any Properties::emptyAny;
+
+Properties::Properties(const ServiceProperties& p)
+{
+  if (p.size() > static_cast<std::size_t>(std::numeric_limits<int>::max()))
+  {
+    throw std::runtime_error("ServiceProperties object contains too many keys");
+  }
+
+  keys.reserve(p.size());
+  values.reserve(p.size());
+
+  for (ServiceProperties::const_iterator iter = p.begin();
+       iter != p.end(); ++iter)
+  {
+    if (Find(iter->first) > -1)
+    {
+      std::string msg = "ServiceProperties object contains case variants of the key: ";
+      msg += iter->first;
+      throw std::runtime_error(msg.c_str());
+    }
+    keys.push_back(iter->first);
+    values.push_back(iter->second);
+  }
```

Comment:
There looks to be common code shared between the two constructors. I'd suggest refactoring to hoist the common code into one place. This makes maintenance a little easier.